tests/aiohttp/test_query_params.py

```diff
@@ -0,0 +1,71 @@
+async def test_no_graphiql_no_query(aiohttp_app_client):
+    params = {
+        "variables": """
+            query {
+                hello
+            }
+        """
+    }
+
+    response = await aiohttp_app_client.get("/graphql", params=params)
+    assert response.status == 400
+
+
+async def test_no_graphiql_get_with_query_params(aiohttp_app_client):
+    query = {
+        "query": """
+            query {
+                hello
+            }
+        """
+    }
+
+    response = await aiohttp_app_client.get("/graphql", params=query)
+    data = await response.json()
+    assert response.status == 200
+    assert data["data"]["hello"] == "Hello world"
+
+
+async def test_post_fails_with_query_params(aiohttp_app_client):
+    query = {
+        "query": """
+            query {
+                hello
+            }
+        """
+    }
+
+    response = await aiohttp_app_client.post("/graphql", params=query)
+
+    assert response.status == 400
+
+
+async def test_does_not_allow_mutation(aiohttp_app_client):
+    query = {
+        "query": """
+            mutation {
+                hello
+            }
+        """
+    }
+
+    response = await aiohttp_app_client.get("/graphql", params=query)
+    assert response.status == 400
+
+    data = await response.text()
+    assert data == "400: mutations are not allowed when using GET"
+
+
+async def test_fails_if_allow_queries_via_get_false(aiohttp_app_client_no_get):
+    query = {
+        "query": """
+            query {
+                hello
+            }
+        """
+    }
+
+    response = await aiohttp_app_client_no_get.get("/graphql", params=query)
+    assert response.status == 400
```

Comment:
Why does this return a `400` and not a `405`?

Reply:
because the method is supported, but the usage is wrong :)